Firestore/core/src/remote/firebase_metadata_provider_apple.mm

```diff
@@ -51,7 +52,7 @@ FIRHeartbeatInfoCode GetHeartbeat() {
 
 void FirebaseMetadataProviderApple::UpdateMetadata(
     grpc::ClientContext& context) {
-  FIRHeartbeatInfoCode heartbeat = GetHeartbeat();
+  FIRHeartbeatInfoCode heartbeat = GetHeartbeat(app_);
```

Comment:
Is the comment below affected?

Reply:
Good question. I'm not sure I understand the context behind the comment because the current code looks to be optimized to _not_ include `FIRHeartbeatInfoCodeNone` in request when the code is `FIRHeartbeatInfoCodeNone`. 

But I dont understand why this matters:
```
 // This should only be changed once it's possible to notify the heartbeat that the 
 // previous attempt to send it has failed. 
```

I'm leaning towards removing the comment. 

@schmidt-sebastian, do you have any context here?

https://github.com/firebase/firebase-ios-sdk/blob/0386388084f1851321ba29bf7d6b3c505bb5f489/Firestore/core/src/remote/firebase_metadata_provider_apple.mm#L55-L62

Reply:
I suspect what the comment is after is that we do want to re-send the heartbeat if the previous attempt failed. It seems like we now assume that any heartbeat attempt succeeded. Once we know that the attempt failed we can send it again, but I am not sure if this code exists yet.

Reply:
I see, this retry behavior is not yet implemented and there are no plans to as of right now. 

I think I will remove the comment because if retry behavior is added, all integrators will be updated accordingly.

Reply:
The comment is saying "If we implement this behavior we need to fix this line of code" so it is a good reminder that there is code here that needs to be changed. It would however helped if this wasn't a TODO for an ex-teammember anymore.

Reply:
SGTM, Ill adjust accordingly.

Reply:
Reworded in [721f74d](https://github.com/firebase/firebase-ios-sdk/pull/9555/commits/721f74d5386b1bde21f988270375a4a8e331a576)